storage/innobase/srv/srv0srv.cc

```diff
@@ -2145,6 +2153,9 @@ srv_master_do_active_tasks(void)
 		return;
 	}
 
+	srv_main_thread_op_info = "doing background file truncate";
+	row_truncate_file_for_mysql_in_background();
```

Comment:
I would do this as part of the function that handles background drop table.
First, it would finish truncating the current file (if any), and once done, it would pick the next element from the drop table queue.
Note: this would not block further `RENAME` or `CREATE` operations, because in MariaDB, we’d have renamed the tables to names starting with `#sql-ib` before adding them to the queue.